### PR TITLE
feat: refactor TLS and server

### PR DESCRIPTION
**Pull request by sethyates**
Refactor TLS support into interfaces and remove disused functions.

## Description
Cleanup TLS config

## Related Issue
#7

## Motivation and Context
First step in refactoring of Server setup.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tls_credentials.go [hunk 1]
```diff
@@ -110,7 +111,10 @@ func NewClientTLSConfig(cfg HasTLSConfig, vault HasVaultConfig) (*tls.Config, er
 
 	var err error
 	t := CloneTLSConfig(cfg)
-	t.Cert.Secret, t.Key.Secret, err = GenerateCertificates(t.Generate, vault)
+
+	certGenerator := NewCertificateGenerator(cfg.GetGenerate(), vault)
+
+	err = certGenerator.GenerateCertificates(&t.Cert.Secret, &t.Key.Secret)
```
Comment (smalexan) [inline tls_credentials.go:117]: 👍

File: key.go [hunk 2]
```diff
@@ -32,26 +32,6 @@ import (
 	"io/ioutil"
 )
 
-// LoadPrivateKeyFromVault loads public key material from vault
-func LoadPrivateKeyFromVault(cfg HasKeyConfig, vault HasVaultConfig) (*rsa.PrivateKey, error) {
```
Comment (smalexan) [inline key.go:36]: Called in orlop-auth

File: key.go [hunk 3]
```diff
@@ -32,26 +32,6 @@ import (
...
-	key, err := LoadKey(cfg, vault, "private")
-	if err != nil {
-		return nil, err
-	}
-
-	return LoadPrivateKey(key)
-}
-
-// LoadPublicKeysFromVault loads public key material from vault
-func LoadPublicKeysFromVault(cfg HasKeyConfig, vault HasVaultConfig) ([]*rsa.PublicKey, error) {
```
Comment (smalexan) [inline key.go:46]: Called in orlop-auth

Comment (sethyates) [reply key.go:46]: Moved to orlop-auth...

File: key.go (hunk 2 above)
Comment (sethyates) [reply key.go:36]: Moved to orlop-auth...

**Pull request merged**